convert2rhel/actions/report.py

```diff
@@ -213,7 +216,11 @@ def summary(results, include_all_reports=False, disable_colors=False):
         highest ones.
     :type include_all_reports: bool
     """
-    logger.task("Pre-conversion analysis report")
+    if post_conversion:
+        logger.task("Post-conversion report")
+    else:
+        logger.task("Pre-conversion analysis report")
+
```

Comment:
I feel like this is an antipattern as we have two different uses in one function. Despite the large overlap it feels like we should either make this into a class that just generates stuff easier or we just split out the functionality more and have a higher level function for pre-conversion and post-conversion with logs

Reply:
I would say different purpose rather than uses here... The usability of the function is still the same despite the logging that is being shown at the beginning/end of the function. 

Having a higher level function does not seem too overkill for such a simple if statement to redirect the logging? We would essentially move this if statement to be an outside statement inside (and around) the main function to just call the high level function that will contain basically a log message that will be different from the other...

For example (I'm visualizing the higher level functions like this): 

```python
def pre_ponr_summary(results, include_all_reports=False, disable_colors=False):
    logger.task("Pre-conversion analysis report")
    summary(results, include_all_reports, disable_colors)

def post_ponr_summary(results, include_all_reports=False, disable_colors=False):
    logger.task("Post-conversion analysis report")
    summary(results, include_all_reports, disable_colors)
```

Which in turn, would move the if statement of a pre(post)-conversion report to something like: 

```python
           if post_conversion:
                report.post_ponr_summary(...)
            else:
                report.pre_ponr_summary(...)
 ```
 
Maybe I'm just overlooking this? Do you see it in a simpler way, @SpyTec?

Reply:
I like the idea of a class, though... We could encapsulate also the `summary_as_json` as `summary_as_txt` inside the report... And with that, we could have a single interface that output different types of summary as we need...

Reply:
Okay, I will create a Jira ticket to transform it in a class, trying to describe the best I can the idea behind it. 

I think moving all of this to a class will help make it more organized and extensible as well. I will implement the higher level functions for now.

Reply:
Thanks for the ideas, @SpyTec